### PR TITLE
Improvements to the port of crm_claim_rma

### PR DESCRIPTION
**Pull request by LeartS**

### CODE REVIEW

File: crm_claim_rma/models/crm_claim_rma.py [hunk 1]
```diff
@@ -251,19 +245,15 @@ def _warranty_limit_values(self, invoice, claim_type, product, claim_date):
 
     def set_warranty_limit(self):
```
Comment (mistotebe) [inline crm_claim_rma/models/crm_claim_rma.py:246]: Use @api.one or self.ensure_one()?

File: crm_claim_rma/models/crm_claim_rma.py [hunk 2]
```diff
@@ -371,10 +360,10 @@ def _warranty_return_address_values(self, product, company, warehouse):
...
             return_address_id = return_address.id
             return_type = 'company'
-        location_dest_id = self.get_destination_location(product, warehouse)
+        location_dest = self.get_destination_location(product, warehouse)
         return {'warranty_return_partner': return_address_id,
                 'warranty_type': return_type,
-                'location_dest_id': location_dest_id}
+                'location_dest_id': location_dest.id}
 
     def set_warranty_return_address(self):
```
Comment (mistotebe) [inline crm_claim_rma/models/crm_claim_rma.py:373]: Use @api.one or self.ensure_one()?

File: crm_claim_rma/wizards/claim_make_picking.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        partner_id = self.env.context.get('partner_id')
+
+        if picking_type == 'out' and partner_id:
+            return self.env['res.partner'].browse(
+                partner_id).property_stock_customer
+
+        if picking_type == 'in' and partner_id:
+            # Add the case of return to supplier !
+            lines = self._default_claim_line_ids()
+            return self._get_common_dest_location_from_line(lines)
```
Comment (mistotebe) [inline crm_claim_rma/wizards/claim_make_picking.py:84]: Does this mean you sometimes return an id and sometimes a browse object? Maybe worth making the `_get_common_...` functions return a browse object?

**Comment by LeartS** [comment]
@mistotebe I should have addressed your remarks, thanks.

The PR is ready for review. Keep in mind that I refactored and cleaned up the code without really knowing what it's supposed to do functionally, so I refactored while keeping the code the same functionally.
I cherry-picked some of the commits and changes mentioned here: https://github.com/OCA/rma/pull/32#issuecomment-127954640 and authored / rewrote others.

**Comment by nhomar** [comment]
@aristobulo-eezee can you merge in order to make the review properly in the main pr to oca/rma. 

Thanks a lot to both of you......

**Comment by nhomar** [comment]
wao.... that was fast
jejejeeje

**Comment by LeartS** [comment]
Yeah, it's better to review there, thanks a lot. Very efficient community :)

**Pull request merged**